File: 01-java-puro/src/main/java/com/tqi/checkout/domain/pix/PixPagamento.java
package com.tqi.checkout.domain.pix;
import com.tqi.checkout.domain.MetodoPagamento;

// [🟨L] - LISKOV SUBSTITUTION PRINCIPLE: Substitui a interface mãe perfeitamente sem quebras de comportamento.
public class PixPagamento implements MetodoPagamento {
    @Override
    public void processar(double valor) {
        System.out.println("⚡PIX: Gerando QR Code no valor de R$ " + valor);
    }
}

File: 02-java-spring-boot/src/main/kotlin/com/tqi/checkout/domain/strategy/PixPagamento.kt
package com.tqi.checkout.domain.strategy
import com.tqi.checkout.domain.MetodoPagamento
import org.springframework.stereotype.Component
@Component
class PixPagamento : MetodoPagamento {
    override fun processar(valor: Double) { println("Spring + Kotlin PIX: R$ $valor") }
}
